### PR TITLE
Events and misc

### PR DESCRIPTION
**Pull request by DamienCassou**

### CODE REVIEW

File: src/widget.js [hunk 1]
```diff
@@ -93,8 +93,15 @@ define(
 				// When within an update transaction, do not update the widget
 				my.inUpdateTransaction = false;
 				children = [];
+				my.initializeSubwidgets(spec);
```
Comment (BenjaminVanRyseghem) [inline src/widget.js]: use `basicInitialize` instead

File: src/events.js [hunk 2]
```diff
@@ -12,7 +12,10 @@ define([],
         var event = function(spec, my) {
             my = my || {};
 
-            var that = function(callback) {
+			// DEPRECATED: use that.on() instead.
+			var that = function(callback) {
```
Comment (BenjaminVanRyseghem) [inline src/events.js]: lovely indentation...

File: src/widget.js (hunk 1 above)
Comment (BenjaminVanRyseghem) [inline src/widget.js]: could be good  to add a regression test to prevent further initialization order issues

File: src/events.js (hunk 2 above)
Comment (DamienCassou) [inline src/events.js]: Agree. This is in the scope of https://github.com/foretagsplatsen/widgetjs/issues/39

**Comment by DamienCassou** [comment]
@BenjaminVanRyseghem I took care of your comments.

File: test/widgetTest.js [hunk 3]
```diff
@@ -342,6 +342,23 @@ define(["widgetjs/widget", "widgetjs/htmlCanvas", "jquery", "chai"], function(wi
 			expect(counter).toBe(1);
 		});
 
+		it("widgets initialize their subwidgets after themselves", function() {
+			var initializationIsDone = false;
+
+			var mySubclass = widget.subclass(function(that, my) {
+				my.initialize = function(spec) {
+					my.super(spec);
+					initializationIsDone = true;
```
Comment (BenjaminVanRyseghem) [inline test/widgetTest.js]: if only spies existed, so you could check when a method is invoked, that would be kinda great...

**Comment by DamienCassou** [comment]
Fixed all your comments.

File: src/widget.js [hunk 4]
```diff
@@ -94,6 +95,17 @@ define(
 				children = [];
 			};
 
+			my.basicInitialize = function(spec) {
```
Comment (BenjaminVanRyseghem) [inline src/widget.js]: shouldn't you use `pre/post` hooks instead?

File: src/widget.js (hunk 1 above)
Comment (DamienCassou) [inline src/widget.js]: done

File: test/widgetTest.js (hunk 3 above)
Comment (DamienCassou) [inline test/widgetTest.js]: done

File: src/widget.js (hunk 4 above)
Comment (DamienCassou) [inline src/widget.js]: done

**Comment by DamienCassou** [comment]
I took care of all your comments.

**Pull request merged**